flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1542,10 +1577,10 @@ def test_no_delete_if_leased(self):
 
     def test_deleted_dataset_volume_unmounted(self):
```

Comment:
Maybe? I think perhaps only with the test you commented on below. I'm not sure if I'm smart enough to disentangle the extraneous details to see which tests are necessary. These tests will all be superfluous once the code is refactored, so I'm inclined to leave them in for the moment.